autometacal/python/metacal.py

```diff
@@ -81,13 +81,26 @@ def generate_mcal_image(gal_images,
 
   # Compute inverse Fourier transform
   img = tf.math.real(tf.signal.fftshift(im_reconv))
+  
+  img=img[:,fact*nx:-fact*nx,fact*ny:-fact*ny]
+
+  return img
+
+def generate_fixnoise(shape,psf_image,reconvolution_psf_image,g):
+  noise = tf.random.normal(shape)
```

Comment:
Is it on purpose that your noise variance is fixed to 1? Might be good for testing if you always set the noise variance to 1 but it will failed in a more general context.

Reply:
arf, yeah ^^ that's no good. In practice, I think it's reasonable to provide the function with a noise realisation instead of letting the function generating the noise itself

Reply:
that's what would happen in a real pipeline as the noise realisation can be complicated

Reply:
In ngmix it is generated from the weight image assuming weights are `1/variance`. And then they generate the noise image.

(Which is why they have the first rotation)

Reply:
That was an overlook. I was testing like this to change it after and forgot to do the rest. Thanks! I'll use noise realisations as an outside input to the function.